migration/v0.40/prepare/20-update-google-dex-config.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+tmp_secrets=$(mktemp)
+append_trap "rm ${tmp_secrets}" EXIT
+
+sed "s/^\([ \t]*\)adminEmail: \(.*\)$/\1domainToAdminEmail: \n\1  \'\*\': \2/" <(sops -d "${CK8S_CONFIG_PATH}/secrets.yaml") | sops -e --input-type yaml --config "${CK8S_CONFIG_PATH}/.sops.yaml" /dev/stdin > "${tmp_secrets}"
```

Comment:
This doesn't seem to work for me with sops 3.8.1

> Could not marshal tree: Error marshaling to json: invalid character '}' looking for beginning of object key string

Moving the `--input-type yaml` until after `--config .../.sops.yaml` made it encrypt the whole thing as a single blob

Could this not be accomplished with `sops --extract` and friends?

Reply:
I can try to do something with that instead. I think I would have to extract and then set the whole dex.connectors block

Reply:
Updated the script now, please try this one as well.